sale_stock_picking_blocking/models/sale_order.py

```diff
@@ -26,9 +26,9 @@ def _check_not_auto_done(self):
             )
 
     @api.onchange("partner_id")
-    def onchange_partner_id(self):
+    def _onchange_partner_id_warning(self):
         """Add the 'Default Delivery Block Reason' if set in the partner."""
-        res = super().onchange_partner_id()
+        res = super()._onchange_partner_id_warning()
```

Comment:
@sbejaoui  `delivery_block_id` should become a compted field with `readonly=False`. It is important that no more business logic depends on onchange methods to make sure that the logic is applied even if the creation is done by a call to the create method from within the code.